src/Domain/Creators/BinaryCompactObject.hpp

```diff
@@ -0,0 +1,205 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <vector>
+
+#include "Domain/Creators/DomainCreator.hpp"  // IWYU pragma: keep
+#include "Domain/Domain.hpp"
+#include "Options/Options.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace domain {
+namespace creators {
+
+/*!
+ * \ingroup DomainCreatorsGroup
+ *
+ * \brief A general domain for two compact objects.
+ *
+ * \image html binary_compact_object_domain.png "A BHNS domain."
+ *
+ * Creates a 3D Domain consisting of four/five layers of blocks:
+ * - 0: The block at the center of each compact object, if not excised.
+ * - 1: The first layer, has the blocks for each compact object.
+ * - 2: The second layer, has the blocks that surround each object with a cube.
+ * - 3: The third layer, has the blocks that surround each cube with a
+ * half-cube.
+ * - 4: The fourth layer is made of the blocks that form the outer sphere.
+ *
+ * `ObjectA` and `ObjectB` refer to the Shell or Sphere making up each compact
+ * object.
```

Comment:
I suggest,

"In the code and options below, `ObjectA` and `ObjectB` refer to the two compact objects, and by extension, also refer to the layers that immediately surround each compact object. Note that `ObjectA` is located to the left of the origin <perhaps specify: in blah coordinates?>, and `ObjectB` is located to the right of the origin."